photon-ml/src/main/scala/com/linkedin/photon/ml/data/AvroDataReader.scala

```diff
@@ -164,6 +164,7 @@ class AvroDataReader(
     // Infer sql schema from avro schema
     val srcFeatureCols = featureColumnMap.values.flatten.toSet
     val schemaFields = inferSchemaFields(records)
+      .getOrElse(throw new IllegalArgumentException("No records found at given input paths."))
```

Comment:
I was going to do that, but at this point the path is not available, and it would require some non-trivial reorganization of the code to surface it. Also we generally log the paths before passing them to this function.